Added Eldritch_Edge and Blighting_Brand NMs and stuff 2.0

Fiocitrine opened the pull request:
Ok so I didn't push a bunch of stuff this time with my added stuff. Sorry for the annoyance.

xdemolish commented:
amend > descriptive commit message > force push

drop a comment when done

TeoTwawki commented on scripts/zones/Rolanberry_Fields/mobs/Eldritch_Edge.lua:
```diff
@@ -0,0 +1,24 @@
+    local PH = GetServerVariable("[PH]Eldritch_Edge");
+        SetServerVariable("[PH]Eldritch_Edge", 0);
+        DeterMob(PH, false);
+   		GetMobByID(PH):setRespawnTime(GetMobRespawnTime(PH));
```
Taaaaaaaaaaaaaaabs

TeoTwawki commented on scripts/zones/Sauromugue_Champaign/mobs/Blighting_Brand.lua:
```diff
@@ -11,4 +11,15 @@ require("scripts/globals/fieldsofvalor");
+    local PH = GetServerVariable("[PH]Blighting_Brand");
+        SetServerVariable("[PH]Blighting_Brand", 0);
+        DeterMob(PH, false);
+   		GetMobByID(PH):setRespawnTime(GetMobRespawnTime(PH));
```
Tabs again

Fiocitrine commented:
fixed dem tabs and alignment stuffs. also fixed Lamina and hyakinthos drop rates.

Fiocitrine commented:
Fixed that indent, love.

xdemolish commented:
see my previous comment

Fiocitrine commented:
"More recent "master" branch build has been stated." What does this mean?

xdemolish commented on scripts/zones/Rolanberry_Fields/mobs/Evil_Weapon.lua:
```diff
@@ -0,0 +1,27 @@
+    local mobID = mob:getID();
+    if (Eldritch_Edge_PH[mobID] ~= nil) then
+        local ToD = GetServerVariable("[POP]Eldritch_Edge");
+        if (ToD <= os.time(t) and GetMobAction(Eldritch_Edge) == 0) then
```
t is undefined

xdemolish replied:
os.time() will do

TeoTwawki replied:
t is not undefined, its part of `os.time()`'s internal table:
http://www.lua.org/manual/5.1/manual.html#pdf-os.time
`t` tells it "give me the clock in seconds" 
https://github.com/DarkstarProject/darkstar/blob/0307e011f10bb108b6bbc6bf81731ea9f3e7608d/scripts/zones/Korroloka_Tunnel/mobs/Clipper.lua#L20
https://github.com/DarkstarProject/darkstar/blob/0307e011f10bb108b6bbc6bf81731ea9f3e7608d/scripts/zones/Korroloka_Tunnel/mobs/Cargo_Crab_Colin.lua#L14
And is correct here.

xdemolish commented on scripts/zones/Sauromugue_Champaign/mobs/Evil_Weapon.lua:
```diff
@@ -10,5 +10,19 @@ require("scripts/globals/fieldsofvalor");
+    local mobID = mob:getID();
+    if (Blighting_Brand_PH[mobID] ~= nil) then
+        local ToD = GetServerVariable("[POP]Blighting_Brand");
+        if (ToD <= os.time(t) and GetMobAction(Blighting_Brand) == 0) then
```
same here

TeoTwawki replied:
still correct here

xdemolish commented:
still want a descriptive commit message, though

Fiocitrine commented:
Ok, I can do that. But I'm still unsure what you mean by "descriptive."

teschnei commented:
your commit message says "Gah noobs"

There is no way you think that is descriptive

Fiocitrine commented:
Of course not. I wasn't sure for what you were looking, exactly. Considering you guys are particular about what goes into the database, that's all.

Fiocitrine commented:
I added the NMs Eldritch Edge and Blighting Brand and a way to spawn them. Also gave a spell list for Blighting Brand. Fixed the Lamina and Hyakinthos droprate for their items to reflect retail.

xdemolish commented:
rebase / squash > use that shiny new descriptive commit message > force push

Fiocitrine commented:
I'll fix that.

teschnei commented:
okay, I thought you were thinking the pull request name was the commit message :p

the pull request name is pretty good - just like "Added Eldritch Edge and Blighting Brand NM" is fine
The commit message should just say what was done - you can check the messages for any of mine to see examples

Fiocitrine commented:
Ok there we go. I rebased and it renamed the commit? Things are alright, for now. 

I added the NMs Eldritch Edge and Blighting Brand and a way to spawn them. Also gave a spell list for Blighting Brand. Fixed the Lamina and Hyakinthos droprate for their items to reflect retail.

xdemolish commented on scripts/zones/Rolanberry_Fields/mobs/Eldritch_Edge.lua:
```diff
@@ -0,0 +1,24 @@
+
+    -- Set Eldritch Edge's Window Open Time
+    local wait = math.random(5400,7200);
+    SetServerVariable("[POP]Eldritch_Edge", os.time(t) + wait); -- 90 to 120 minutes
```
whoops missed this one

xdemolish commented on scripts/zones/Sauromugue_Champaign/mobs/Blighting_Brand.lua:
```diff
@@ -11,4 +11,15 @@ require("scripts/globals/fieldsofvalor");
+
+     -- Set Eldritch Edge's Window Open Time
+    local wait = math.random(5400,7200);
+    SetServerVariable("[POP]Blighting_Brand", os.time(t) + wait); -- 90 to 120 minutes
```
os.time() here too

TeoTwawki commented:
@xdemolish @fiocitrine `os.time(t)` was in fact correct. Its used this way all throughout existing scripts and is not undefined - its specifies which table os.time() is to return, in this case "clock in seconds".

xdemolish commented:
wait shit i thought that was 't' not just t

TeoTwawki commented:
Wtf github eating some of my comments

@xdemolish I had linked you the lua book and 2 instances of its use in DSP that just use t, github seems to have eaten that post. I dunno anything about 't' instead of t but if the DSP example I liked is wrong then most and DSP is wrong & lol at that.

Fiocitrine commented:
Should I change it back? I'm confused...

Fiocitrine commented:
Well I'm not getting this mob to spawn, and it worked with (t) so I'm just gonna set it back to that.
